backend-rust/src/indexer.rs

```diff
@@ -1293,298 +1176,353 @@ impl PreparedBlockItem {
 }
 
 /// Different types of block item events that can be prepared.
-enum PreparedEvent {
+enum PreparedBlockItemEvent {
     /// A new account got created.
     AccountCreation(PreparedAccountCreation),
+    /// An account transaction event.
+    AccountTransaction(Box<PreparedAccountTransaction>),
+    /// Chain update transaction event.
+    ChainUpdate,
+}
+
+impl PreparedBlockItemEvent {
+    async fn prepare(
+        node_client: &mut v2::Client,
+        data: &BlockData,
+        item_summary: &BlockItemSummary,
+        item: &BlockItem<EncodedPayload>,
+    ) -> anyhow::Result<Self> {
+        match &item_summary.details {
+            BlockItemSummaryDetails::AccountCreation(details) => Ok(
+                PreparedBlockItemEvent::AccountCreation(PreparedAccountCreation::prepare(details)?),
+            ),
+            BlockItemSummaryDetails::AccountTransaction(details) => {
+                let sender = Arc::new(details.sender.to_string());
```

Comment:
I see no value in this extra step. I would just pass the string

Reply:
As mentioned in another comment, this ensures that we only allocate the string representation of the sender once in memory and the rest just refer to this same location.